icicle_v3/backend/cpu/src/field/cpu_vec_ops.cpp

```diff
@@ -66,23 +338,36 @@ template <typename T>
 eIcicleError
 cpu_vector_div(const Device& device, const T* vec_a, const T* vec_b, uint64_t n, const VecOpsConfig& config, T* output)
 {
-  for (uint64_t i = 0; i < n; ++i) {
-    output[i] = vec_a[i] * T::inverse(vec_b[i]);
-  }
-  return eIcicleError::SUCCESS;
+  return cpu_2vectors_op(VecOperation::VECTOR_DIV, vec_a, vec_b, n, config, output);
 }
 
 REGISTER_VECTOR_DIV_BACKEND("CPU", cpu_vector_div<scalar_t>);
 
+/*********************************** SUM ***********************************/
+template <typename T>
+eIcicleError cpu_vector_sum(const Device& device, const T* vec_a, uint64_t n, const VecOpsConfig& config, T* output)
+{
+  return cpu_1vector_op(VecOperation::VECTOR_SUM, vec_a, n, config, output);
```

Comment:
Is this computing the sum of the vector? I may be wrong but It seems you don't reduce to one value.